Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -534,9 +541,15 @@ void GraphicsContextSkia::setLineCap(LineCap lineCap)
     m_skiaState.m_stroke.cap = toSkiaCap(lineCap);
 }
 
-void GraphicsContextSkia::setLineDash(const DashArray&, float /*dashOffset*/)
+void GraphicsContextSkia::setLineDash(const DashArray& dashArray, float dashOffset)
 {
-    notImplemented();
+    ASSERT(!(dashArray.size() % 2));
+
+    if (dashArray.size() > 0)
+        m_skiaState.m_dash.array = std::optional<DashArray>(dashArray);
+    else
+        m_skiaState.m_dash.array = std::nullopt;
+    m_skiaState.m_dash.offset = dashOffset;
```

Comment:
```suggestion
    m_skiaState.m_dash.array = WTFMove(dashArray);
    m_skiaState.m_dash.offset = dashOffset;
```

Reply:
Assuming we can change the method to use move semantics.

Reply:
This method is declared in the parent class (`GraphicsContext`) and it seems like changing that will involve changing various other source files. I can do that if you think it's worth it.

Reply:
The implementation of `BifurcatedGraphicsContext::setLineDash` seems to get in the way of changing the method to use move semantics. It proxies the array to two other contexts, and neither of them can move the array on its own, I think?

Reply:
Ok its fine then.

Reply:
Hm, as I go through this, I feel like changing this semantic would be better off as a separate patch. It's touching quite a lot of files, and the diff is growing disproportionately compared to the actual dashed-lines implementation. Do you think it's okay to do that separatedly?

Reply:
I didn't see your response, sorry. I'll resolve these threads. Thanks again for the reviews!